targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/trng_api.c

```diff
@@ -52,44 +52,58 @@
 #define TRNG_CNT_VAL	4095
 #define TRNG_PRESCALER 	2
 
-static ADI_RNG_HANDLE     RNGhDevice;                     /* Memory to handle CRC Device */
-static uint32_t           RngDevMem[(ADI_RNG_MEMORY_SIZE + 3)/4]; /* Data buffers for Random numbers */
+/* Data buffers for Random numbers */
+static uint32_t RngDevMem[(ADI_RNG_MEMORY_SIZE + 3)/4];
 
 void trng_init(trng_t *obj)
 {
-    (void)obj;
+	ADI_RNG_HANDLE RNGhDevice;
+
+	// Open the device
     adi_rng_Open(0,RngDevMem,sizeof(RngDevMem),&RNGhDevice);
 
     // Set sample length for the H/W RN accumulator
     adi_rng_SetSampleLen(RNGhDevice, TRNG_PRESCALER, TRNG_CNT_VAL);
 
-    // Enable the RNG
+	// Disable buffering - single byte generation only
+	adi_rng_EnableBuffering(RNGhDevice, false);
+
+	// Enable the TRNG
     adi_rng_Enable(RNGhDevice, true);
+
+	// Save device handle
+	obj->RNGhDevice = RNGhDevice;
```

Comment:
`RNGhDevice` is located on the stack. when you assign `obj->RNGhDevice = RNGhDevice` you set `obj->RNGhDevice` to be point to a variable on the stack, that might be garbage after exiting the function. Does `adi_rng_Open` allocate memory on the heap?

Reply:
Do all the `adi_xxx` functions return an error code? WHat happens if they fail?

Reply:
@dave-wu  Please address the first comment\question that was hidden, It is still relevent